### PR TITLE
[Madara] Decrypt image list for Amuy and Cerise Scans

### LINKED ISSUE #14963: Cerise Scans: No pages found
### Source information and language

Cerise scans BR 1.3.30

### Steps to reproduce

Screenshot_2023-01-15-15-17-42-026_eu kanade tachiyomi

### Expected behavior

Be able to read and download the manga

### Actual behavior

It is not possible to download or read the mangas.

### Tachiyomi version

0.14.3

### Android version

10

### Other details

When I try to read it, it says there are no pages.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.14.3](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #15071: Amuy Scan: No pages found
### Source information and language

Amuy 1.3.27

### Steps to reproduce

Can't find any page

### Expected behavior

Open the chapter normally with the pages

### Actual behavior

Can't find any page

### Tachiyomi version

0.14.3

### Android version

Android 11

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.14.3](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by beerpiss**
Closes #15071
Closes #14963 

Checklist:

- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [x] Have explicitly kept the `id` if a source's name or language were changed
- [ ] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

**Comment by beerpiss** [comment]
Actually I'm not even sure if this is mergeable since it fails in CI due to a dependency on `:lib-cryptoaes`.

**Comment by curche** [comment]
Maybe related to #13860

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/madara/Madara.kt [hunk 1]
```diff
@@ -1014,17 +1019,45 @@ abstract class Madara(
...
+            return document.select(pageListParseSelector).mapIndexed { index, element ->
+                Page(
+                    index,
+                    document.location(),
+                    element.select("img").first()?.let {
+                        it.absUrl(if (it.hasAttr("data-src")) "data-src" else "src")
+                    }
+                )
+            }
         }
```
Comment (AntsyLich) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/madara/Madara.kt:1061]:
```suggestion
        if (chapterProtector == null) {
            return document.select(pageListParseSelector).mapIndexed { index, element ->
                val imageUrl = element.selectFirst("img")?.let { imageFromElement(it) }
                Page(index, document.location(), imageUrl)
            }
        }
        val chapterProtectorHtml = chapterProtector.html()
            val password = chapterProtectorHtml
                .substringAfter("wpmangaprotectornonce='")
                .substringBefore("';")
            val chapterData = json.parseToJsonElement(
                chapterProtectorHtml
                    .substringAfter("chapter_data='")
                    .substringBefore("';")
                    .replace("\\/", "/")
            ).jsonObject

            val unsaltedCiphertext = Base64.decode(chapterData["ct"]!!.jsonPrimitive.content, Base64.DEFAULT)
            val salt = chapterData["s"]!!.jsonPrimitive.content.decodeHex()
            val ciphertext = "Salted__".toByteArray(Charsets.UTF_8) + salt + unsaltedCiphertext

            val rawImgArray = CryptoAES.decrypt(Base64.encodeToString(ciphertext, Base64.DEFAULT), password)
            val imgArrayString = json.parseToJsonElement(rawImgArray).jsonPrimitive.content
            val imgArray = json.parseToJsonElement(imgArrayString).jsonArray

            return imgArray.mapIndexed { idx, it ->
                Page(idx, document.location(), it.jsonPrimitive.content)
            }
```

**Comment by beerpiss** [comment]
I wanted to make chapter decryption template-wide, but I guess that is currently not possible due to limitations in the `multisrc` build system.

Changed approach to adding chapter decryption to affected extensions (for now).

**Pull request merged**